analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml

```diff
@@ -20,111 +20,111 @@ project:
-          - id: "Cargo::winapi-i686-pc-windows-gnu:0.4.0"
+          - id: "Crate::winapi-i686-pc-windows-gnu:0.4.0"
             linkage: "STATIC"
-          - id: "Cargo::winapi-x86_64-pc-windows-gnu:0.4.0"
+          - id: "Crate::winapi-x86_64-pc-windows-gnu:0.4.0"
             linkage: "STATIC"
-      - id: "Cargo::rand_os:0.1.3"
+      - id: "Crate::rand_os:0.1.3"
         linkage: "STATIC"
         dependencies:
-        - id: "Cargo::cloudabi:0.0.3"
+        - id: "Crate::cloudabi:0.0.3"
           linkage: "STATIC"
           dependencies:
-          - id: "Cargo::bitflags:1.1.0"
+          - id: "Crate::bitflags:1.1.0"
             linkage: "STATIC"
-        - id: "Cargo::fuchsia-cprng:0.1.1"
+        - id: "Crate::fuchsia-cprng:0.1.1"
           linkage: "STATIC"
-        - id: "Cargo::libc:0.2.62"
+        - id: "Crate::libc:0.2.62"
           linkage: "STATIC"
-        - id: "Cargo::rand_core:0.4.2"
+        - id: "Crate::rand_core:0.4.2"
           linkage: "STATIC"
-        - id: "Cargo::rdrand:0.4.0"
+        - id: "Crate::rdrand:0.4.0"
           linkage: "STATIC"
           dependencies:
-          - id: "Cargo::rand_core:0.3.1"
+          - id: "Crate::rand_core:0.3.1"
             linkage: "STATIC"
             dependencies:
-            - id: "Cargo::rand_core:0.4.2"
+            - id: "Crate::rand_core:0.4.2"
               linkage: "STATIC"
-        - id: "Cargo::winapi:0.3.8"
+        - id: "Crate::winapi:0.3.8"
           linkage: "STATIC"
           dependencies:
-          - id: "Cargo::winapi-i686-pc-windows-gnu:0.4.0"
+          - id: "Crate::winapi-i686-pc-windows-gnu:0.4.0"
             linkage: "STATIC"
-          - id: "Cargo::winapi-x86_64-pc-windows-gnu:0.4.0"
+          - id: "Crate::winapi-x86_64-pc-windows-gnu:0.4.0"
             linkage: "STATIC"
-      - id: "Cargo::rand_pcg:0.1.2"
+      - id: "Crate::rand_pcg:0.1.2"
         linkage: "STATIC"
         dependencies:
-        - id: "Cargo::rand_core:0.4.2"
+        - id: "Crate::rand_core:0.4.2"
           linkage: "STATIC"
-      - id: "Cargo::rand_xorshift:0.1.1"
+      - id: "Crate::rand_xorshift:0.1.1"
         linkage: "STATIC"
         dependencies:
-        - id: "Cargo::rand_core:0.3.1"
+        - id: "Crate::rand_core:0.3.1"
           linkage: "STATIC"
           dependencies:
-          - id: "Cargo::rand_core:0.4.2"
+          - id: "Crate::rand_core:0.4.2"
             linkage: "STATIC"
-      - id: "Cargo::winapi:0.3.8"
+      - id: "Crate::winapi:0.3.8"
         linkage: "STATIC"
         dependencies:
-        - id: "Cargo::winapi-i686-pc-windows-gnu:0.4.0"
+        - id: "Crate::winapi-i686-pc-windows-gnu:0.4.0"
           linkage: "STATIC"
-        - id: "Cargo::winapi-x86_64-pc-windows-gnu:0.4.0"
+        - id: "Crate::winapi-x86_64-pc-windows-gnu:0.4.0"
           linkage: "STATIC"
 packages:
-- id: "Cargo::autocfg:0.1.6"
-  purl: "pkg:cargo/autocfg@0.1.6"
+- id: "Crate::autocfg:0.1.6"
+  purl: "pkg:crate/autocfg@0.1.6"
```

Comment:
The [PURL spec specifies](https://github.com/package-url/purl-spec#known-purl-types) that for Rust packages `cargo` should be used.
As you're mapping `"crate" -> PurlType.CARGO` we're now handling 2 different types of PURLs.
Not sure how to handle that, because the PURL spec is inconsistent in this regard.

Reply:
Thanks, I indeed need to keep the expected PURLs here as-is. But that is actually due to a bug in the PURL spec, IMO, also see https://github.com/package-url/purl-spec/pull/75#issuecomment-737075244.